components/writers/src/main/java/org/datacleaner/beans/writers/DeleteTableRowAnalyzer.java

```diff
@@ -0,0 +1,472 @@
+            }
+        } else if (errorLogFile.isDirectory()) {
+            file = new File(errorLogFile, "updation_error_log.csv");
+        } else {
+            file = errorLogFile;
+        }
+
+        final CsvDataContext dc = new CsvDataContext(file);
+
+        final Schema schema = dc.getDefaultSchema();
+
+        if (file.exists() && file.length() > 0) {
+            validateCsvHeaders(dc);
+        } else {
+            // create table if no table exists.
+            dc.executeUpdate(new UpdateScript() {
+                @Override
+                public void run(UpdateCallback cb) {
+                    TableCreationBuilder tableBuilder = cb.createTable(schema, "error_table");
+                    for (String columnName : conditionColumnNames) {
+                        tableBuilder = tableBuilder.withColumn(columnName);
+                    }
+
+                    tableBuilder = tableBuilder.withColumn(ERROR_MESSAGE_COLUMN_NAME);
+
+                    tableBuilder.execute();
+                }
+            });
+        }
+
+        return dc;
+    }
+
+    @Override
+    public void run(InputRow row, int distinctCount) {
+
+        final Object[] rowData;
+        if (additionalErrorLogValues == null) {
+            rowData = new Object[conditionColumnNames.length];
+        } else {
+            rowData = new Object[conditionColumnNames.length + additionalErrorLogValues.length];
+        }
+        for (int i = 0; i < conditionValues.length; i++) {
+            rowData[i] = row.getValue(conditionValues[i]);
+        }
+
+        if (additionalErrorLogValues != null) {
+            for (int i = 0; i < additionalErrorLogValues.length; i++) {
+                Object value = row.getValue(additionalErrorLogValues[i]);
+                rowData[conditionColumnNames.length + i] = value;
+            }
+        }
+
+        try {
+            // perform conversion in a separate loop, since it might crash and
+            // the
+            // error data will be more complete if first loop finished.
+            for (int i = 0; i < conditionValues.length; i++) {
+                int index = i;
+                rowData[index] = convertType(rowData[index], _targetConditionColumns[i]);
+
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Value for {} set to: {}", conditionColumnNames[i], rowData[index]);
+                }
+            }
+        } catch (RuntimeException e) {
+            for (int i = 0; i < distinctCount; i++) {
+                errorOccurred(rowData, e);
+            }
+            return;
+        }
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("Adding row data to buffer: {}", Arrays.toString(rowData));
+        }
+
+        for (int i = 0; i < distinctCount; i++) {
+            _writeBuffer.addToBuffer(rowData);
+        }
+    }
+
+    private Object convertType(final Object value, Column targetColumn) throws IllegalArgumentException {
```

Comment:
This method (and probably also more) is certainly a candidate for reuse between this class and UpdateTableAnalyzer. I would suggest making a small (non-public) utility class for the two analyzers.